src/js/components/TaskView.js

```diff
@@ -60,6 +61,24 @@ class TaskView extends mixin(SaveStateMixin, StoreMixin) {
     super.componentWillMount();
   }
 
+  componentWillReceiveProps(nextProps) {
+    let {filterByStatus, searchString} = this.state;
+    let tasks = this.getFilteredTasks(filterByStatus, nextProps.tasks, searchString);
+    let prevTasks = this.getFilteredTasks(filterByStatus, this.props.tasks, searchString);
+
+    if (!deepEqual(tasks, prevTasks)) {
```

Comment:
Sorry if I'm missing something, but instead of calling `getFilteredTasks` at the top of this method, can't we just call `deepEqual` with `nextProps.tasks` to `this.props.tasks` in this `if` statement? Then, if that's truthy, we can call `getFilteredTasks` on each set of tasks.